.github/workflows/pull-request.yaml

```diff
@@ -26,7 +26,7 @@ jobs:
         run: |
           test -f .ciignore || wget https://raw.githubusercontent.com/canonical/observability/main/.ciignore
           while IFS= read -r globline; do
-            compgen -G "$globline" >> .ciignore-expanded
+            python -c "from glob import glob; print(\"\n\".join(glob(\"$globline\", recursive=True)))" >> .ciignore-expanded
```

Comment:
Looks ok.
1. What was the issue that prompted this solution?
2. Have you considered `shopt -s extglob`?

Reply:
1. `compgen` was not found in the shell, for some reason, though it's documented to be [builtin](https://manpages.ubuntu.com/manpages/bionic/man7/bash-builtins.7.html).
2. `shopt -s extglob` extends the shell pattern matching behavior, it doesn't do the matching itself, which is what we need